direct/direct.go

```diff
@@ -26,13 +32,24 @@ type directClient struct {
 	responses map[string]chan *types.Envelope
 	m         sync.Mutex
 	twinDB    TwinDB
+	privKey   *secp256k1.PrivateKey
+}
+
+func GenerateSecureKey(mnemonics string) (*secp256k1.PrivateKey, error) {
+	seed, err := bip39.NewSeedWithErrorChecking(mnemonics, "")
+	if err != nil {
+		return nil, errors.Wrap(err, "failed to decode mnemonics")
+	}
+	priv := secp256k1.PrivKeyFromBytes(seed[:32])
+	return priv, nil
+
 }
 
 // id is the twin id that is associated with the given identity.
-func NewClient(ctx context.Context, identity substrate.Identity, url string, session string, twinDB TwinDB) (rmb.Client, error) {
+func NewClient(ctx context.Context, identity substrate.Identity, url string, session string, twinDB TwinDB, privKey *secp256k1.PrivateKey) (rmb.Client, error) {
```

Comment:
Since e2e is opt in mean the privKey can in some cases be nil. So if we passed nil as privKey this will cause the client to panic later when u try to do encrypt/decrypt. 

 The function NewClient now takes a lot of arguments, and it's hard to create. So I suggest the client interface is cleaned up more. May be directly pass the mnemonics/seed directly with the key type? then it's up to the client may be to build both identity and the priv key? (just an idea) 

some more comments:
- ctx is not used, may be should be remove
- there is no way to close the client (may be it should be closing on cancelling the ctx, may be this is why i added ctx at first)